distributed/deploy/spec.py

```diff
@@ -11,6 +11,53 @@
 from ..security import Security
 
 
+class SpecProcess:
+    """ An interface for Scheduler and Worker processes for use in SpecCluster
+
+    Parameters
+    ----------
+    loop:
+        A pointer to the running loop.
+
+    """
+
+    def __init__(self, loop=None):
+        self.address = None
+        self.loop = loop
+        self.lock = asyncio.Lock()
+        self.status = "created"
+
+    def __await__(self):
+        async def _():
+            async with self.lock:
+                if self.status == "created":
+                    await self.start()
+                    assert self.status == "running"
+            return self
+
+        return _().__await__()
+
+    @property
+    def worker_address(self):
+        """ For API compatibility with Nanny. """
+        return self.address
```

Comment:
This attribute seems to only make sense for the worker/nanny type objects.  Maybe this encourages us to have two such interface classes?

Reply:
I guess it feels odd to split things for only this one method though.

Reply:
I agree, it feels odd either way. Having two interfaces seems a bit wasteful though.

Reply:
I've made our use of `worker_address` a bit more permissive.  Hopefully this sidesteps this issue.